packages/mattermost-redux/src/selectors/entities/channels.ts

```diff
@@ -353,6 +353,12 @@ export function getChannelByName(state: GlobalState, channelName: string): Chann
     return getChannelByNameHelper(getAllChannels(state), channelName);
 }
 
+export function getChannelByTeamIdAndChannelName(state: GlobalState, teamId: string, channelName: string): Channel | undefined | null {
+    return Object.values(getAllChannels(state)).find((channel) =>
+        channel.team_id === teamId && channel.name === channelName,
+    );
```

Comment:
Maybe we can use [`getChannelsInTeam`](https://github.com/mattermost/mattermost-webapp/blob/master/packages/mattermost-redux/src/selectors/entities/channels.ts#L101-L103) to make this slightly faster:
```suggestion
    return getChannelsInTeam(state)[teamId]?.find((channel) => channel.name === channelName)
```

Reply:
I experimented this for a bit, but `getChannelsInTeam` returns a `Set`, not an array, and then requires a cast to `Array.from` at which point I feel like any performance gains fade from having to copy the data.

Reply:
Huh, you're right. You could also use the `.values()` iterator, but that will make the code more complex, so nevermind, `getAllChannels` is good!

Reply:
Note that those selectors also might treat DMs/GMs differently since they're outside of a team. I forget if `getChannelsInTeam` is one that includes the visible DMs/GMs or not though